woocommerce/google-listings-and-ads
# API PULL - Add Notices inside GMC Card in Settings

### Changes proposed in this Pull Request:

This PR enables some info Notices inside the GMC card under settings depending on the status of WP API Auth Grant 

This PR does some adjustments as:

- Create a component for the Button for initialize the Auth process
- Create a  success notice and a disable link in the GMC Card when the status is "approved" 
- Create a warning notice and a button to initialize the auth process when the status is "error"
- Adapt some Notice styles creating a component (I used the 10Up styles in https://github.com/woocommerce/google-listings-and-ads/pull/2270/files#diff-3034ade2c4e48a4e528de03c43bc15f68606bf48b4fc7e0537668f3eedb49633)

### Screenshots:

When status saved in `gla_wpcom_rest_api_status`  is empty

https://github.com/woocommerce/google-listings-and-ads/assets/5908855/431e521f-58ba-4c60-91e4-3fbcdb1afdea

When status saved in `gla_wpcom_rest_api_status`  is "approved"

https://github.com/woocommerce/google-listings-and-ads/assets/5908855/c998deef-d20d-420b-93db-005ba32f7c4d

When status saved in `gla_wpcom_rest_api_status`  is "error"

https://github.com/woocommerce/google-listings-and-ads/assets/5908855/99307909-f704-4512-8f14-787b719be2fd

### Detailed test instructions:


1. Checkout this PR and be sure `gla_wpcom_rest_api_status` is empty
2. Go to Settings and see the banner at the top (this banner was not touched, its for the parent PR)
3. Check the GMC card is as always
4. Go to the database and add the `gla_wpcom_rest_api_status`  option with the value "approved"
5. Go to Settings and see there is NO banner at the top 
6. Check the GMC card and see a success message and a link for disable the feature
7. Click on that link
8. See a loading message at the bottom
9. When it finishes see the card as always and the banner at the top
10. Go to the database and add the `gla_wpcom_rest_api_status`  option with the value "error"
11. Go to Settings and see there is NO banner at the top 
12. Check the GMC card and see a warning message and a button for grant access
13. Click to the button, it does the same as the banner button
